regression/pages/studio/course_outline_page.py

```diff
@@ -46,22 +47,31 @@ def add_subsection_with_name(self, text):
         Arguments:
             text (str): The sub section added will be named to this
         """
-        self.q(css='.button-new[data-default-name="Subsection"]').click()
+        subsection_css = '.wrapper-subsection-title.wrapper-xblock-field.' \
+                         'incontext-editor.is-editable.is-editing' \
+                         ' .xblock-field-input.incontext-editor-input'
+
+        self.q(
+            css='.button-new[data-default-name="Subsection"]'
+        ).results[-1].click()
+        self.wait_for_element_visibility(
+            subsection_css, 'subsection is visible'
+        )
         self.wait_for_ajax()
         self.q(
-            css='.incontext-editor-input[value="Subsection"]'
-        ).results[0].send_keys(text)
+            css=subsection_css
+        ).results[-1].send_keys(text)
 
         self.q(css='.content').first.click()
         # Click initiates an ajax call
         self.wait_for_ajax()
 
-    def add_unit(self):
+    def click_add_unit_button(self):
         """
         Adds Unit clicking the unit button of a sub section
         Navigates to Add Components page
         """
-        self.q(css='.button-new[data-default-name="Unit"]').click()
+        self.q(css='.button-new[data-default-name="Unit"]').results[-1].click()
```

Comment:
@benpatterson Its just a preventive measure. Section to which we add the unit appears at the end of page ( appended to the list of currently present sections) and we want to make sure we add unit to the subsection of this section.